cocos/2d/CCSprite.cpp

```diff
@@ -365,6 +365,8 @@ void Sprite::setTexture(const std::string &filename)
 
 void Sprite::setTexture(Texture2D *texture)
 {
+    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP));
```

Comment:
If this needs to be here it should reset its own GLProgramState or otherwise make sure to check that it's not a different shader.